pkg/core/cacheddao.go

```diff
@@ -85,6 +87,25 @@ func (cd *cachedDao) PutUnspentCoinState(hash util.Uint256, ucs *state.UnspentCo
 	return nil
 }
 
+// GetNEP5Balances retrieves NEP5Balances for the acc.
+func (cd *cachedDao) GetNEP5Balances(acc util.Uint160) (*state.NEP5Balances, error) {
+	if bs := cd.balances[acc]; bs != nil {
+		return bs, nil
+	}
+	bs, err := cd.dao.GetNEP5Balances(acc)
+	if err != nil {
+		return nil, err
+	}
+	cd.balances[acc] = bs
```

Comment:
I don't think it's necessary to cache it right here, you can do that in `Put` the same way `AccountState` functions do. Balances are usually taken to modify them and put back, they have a different usage model from contracts that do read-caching because they don't change often. I think it simplifies things a bit.